tests/models/eventModelTest.js

```diff
@@ -0,0 +1,200 @@
+'use strict';
+const {Event, ConnectionEvent, OrderEvent, ProductEvent} = require('../../models/eventModel');
+const chai = require('chai');
+const mongoose = require('mongoose');
+const validator = require("validator");
+const expect = chai.expect;
+chai.use(require('chai-as-promised'));
+const dotenv = require('dotenv');
+dotenv.config({
+    path: './config.env'
+});
+
+before(async function () {
+    const database = process.env.DATABASE.replace(
+        '${MONGO_USERNAME}', process.env.MONGO_USERNAME).replace(
+        '${MONGO_PASSWORD}', process.env.MONGO_PASSWORD).replace(
+        '${MONGO_HOSTNAME}', process.env.MONGO_HOSTNAME).replace(
+        '${MONGO_PORT}', process.env.MONGO_PORT).replace(
+        '${MONGO_DB}', process.env.MONGO_DB_TEST);
+
+    console.log(database);
+
+// Connect the database
+    await mongoose.connect(database, {
+        useNewUrlParser: true,
+        useCreateIndex: true,
+        useFindAndModify: false,
+        useUnifiedTopology: true
+    }).then(con => {
+        console.log('DB connection Successfully!');
+        mongoose.connection.db.dropDatabase(console.log(`${mongoose.connection.db.databaseName} database dropped.`)
+        );
+    });
+
+});
+// 606afdb5aa09d43a84b6181a
+describe('EventModel', function () {
+    beforeEach( function (done) {
+        mongoose.connection.db.dropDatabase(() => {console.log(`${mongoose.connection.db.databaseName} database dropped.`);
+            done();});
+    });
+    describe('Connection event', function () {
+        describe('Missing elements', function () {
+            Object.keys(ConnectionEvent.schema.obj).forEach((key) => {
+                if (ConnectionEvent.schema.obj[key].hasOwnProperty('required')) {
+                    let test = {
+                        ip: "127.0.0.1",
+                        userAgent: "Red Star OS 4",
+                        user: "606afdb5aa09d43a84b6181a",
+                    };
+                    delete test[key];
+                    it('should throws when ' + key + ' is not present', async () => {
+                        await expect(ConnectionEvent.create(test)).to.be.rejectedWith(Error);
+                    });
+                }
+            });
+        });
+        describe('IP', function () {
+            it('should throws when value is not a correct ip', async () => {
+                await expect(ConnectionEvent.create({
+                    ip: "127.0.",
+                    userAgent: "Red Star OS 4",
+                    user: "606afdb5aa09d43a84b6181a"
+                })).to.be.rejectedWith(Error);
+            });
+        });
+        describe('user', function () {
+            it('should throws when value is not a correct id', async () => {
```

Comment:
wich id ? you mean user ?

Reply:
Ref does not check if it is a valid object id for the reference (Require another query). Ref is used with populate to automatically fetch the associated data.